### ISSUE TITLE
[FEATURE REQUEST] FastFoto scanned image stack option

### ISSUE THREAD

**Issue by isaacolsen94** [simulot/immich-go#213]
Hey it would be lovely to have the script stack Epson's FastFoto scanned output files. Currently if you scan a photo it can output 3 files, the original scan, a "corrected" scan, and the backside of the photo if it has writing on it. Its structure looks like this:
specified-image-name.jpg            Original
specified-image-name_a.jpg         Corrected
specified-image-name_b.jpg        Back of Photo

Would it be possible to have the script make the original the top photo and stack the _a/_b files behind it?

**Comment by isaacolsen94** [comment]
I played around with codespace and chatgpt and got something similar to what you have that would stack photos that ended in _a and _b, but I really don't know much about programming, but thought I'd paste it here incase it helps, but I know chatgpt is rarely useful for these kind of tasks.

```
var fastFotoRE = regexp.MustCompile(`^(.*)(_a|_b)(\..*)$`)

func fastFoto(name string) (bool, string, bool) {
    parts := fastFotoRE.FindStringSubmatch(name)
    if len(parts) == 0 {
        return false, "", false
    }
    base := strings.TrimSuffix(parts[1], "_a")
    base = strings.TrimSuffix(base, "_b")
    return true, base, false // Assume fastFoto files are not cover
}

var stackMatchers = []stackMatcher{nexusBurst, huaweiBurst, pixelBurst, samsungBurst, fastFoto}
```

**Comment by simulot** [comment]
Thanks for the suggestion.
Added to the todo list

**Comment by isaacolsen94** [comment]
Awesome, thank you so much!

**Comment by isaacolsen94** [comment]
Hey Simulot, first thanks for all the awesome work you've put into this! I just wanted to follow up and see if this was still on your radar?

**Comment by simulot** [comment]
I'm focused on fixing bugs at the moment

**Comment by isaacolsen94** [comment]
Hey! Sorry to bug about this again, but saw you asking about feature requests in the latest release comments. So I thought I'd ping this again. Tha ka for the awesome work you do!

**Comment by simulot** [comment]
That's the right moment to pop up your request!

**Comment by simulot** [comment]
Could provide a non personal  scan to illustrate the case? 
I would like to check what are EXIF fields
Thnks

**Comment by isaacolsen94** [comment]
Yes, I had to do a google drive link because I didn't have any scans small enough to upload here. Here is the link: https://drive.google.com/drive/folders/1RGZl8f1Yxg7uOFfux_V3DEXmevBuvktN?usp=sharing

**Comment by simulot** [comment]
Received, thank you.  Which one you want as cover for the stack?

**Comment by isaacolsen94** [comment]
The _a version

**Comment by simulot** [comment]
👀 
![image](https://github.com/user-attachments/assets/825177e6-7b40-4b0a-98b4-220e29e9c986)

![image](https://github.com/user-attachments/assets/beda0c12-74fd-4a52-a658-c57bb89709fc)

**Comment by isaacolsen94** [comment]
That looks absolutely amazing! Thank you so much!! That will save me hours of manually stacking!!

**Comment by simulot** [comment]
The feature is available in the alpha 4 release
https://github.com/simulot/immich-go/releases/tag/v0.23.0-alpha4

**Comment by isaacolsen94** [comment]
Hey, first off want to thank you for this feature!! It has saved me DAYs of work! I did notice a slight bug though. It works flawlessly except it seems on files with parentheses in the name. Example: 
Stack fine:
1992-xx-xx - Easter_Arizona - 0069.jpg
1992-xx-xx - Easter_Arizona - 0069_a.jpg
1992-xx-xx - Easter_Arizona - 0069_b.jpg

Don't stack (but do upload):
1992-xx-xx - Easter_Arizona (FF-680W-1200-PR) - 0069.jpg
1992-xx-xx - Easter_Arizona (FF-680W-1200-PR) - 0069_a.jpg
1992-xx-xx - Easter_Arizona (FF-680W-1200-PR) - 0069_b.jpg

**Comment by simulot** [comment]
That's my fate: there is always one more bug ;-)

**Comment by isaacolsen94** [comment]
I did see that RC release and felt bad throwing this one up :P I did a bit more digging and it seems to be less the parentheses and more the space before the 0000 numbers at the end

**Comment by simulot** [comment]
The name pattern recognition is a little bit too strict, requiring a `_` followed by digits then `a` or `b` and the extension.

I'm interesting in how the files are generated, which part is given by the user, and which is given by the scan software.
Can I relax the rule, and simply say  that file names always finish by 4 digits then the extension or `_a` or `_b` and the extension?
